### PR TITLE
[Homework #6] - JS-Part 1 by Max Hirning

### PR DESCRIPTION
**Pull request by Max-Hirning**
Have done all tasks except 8th

### CODE REVIEW

File: homeworks/max-hirning/01-js/index.js [hunk 1]
```diff
@@ -0,0 +1,386 @@
...
+        res[el] = (res[el]) ? res[el]+1 : 1;
+        return res;
+    }, {});
+
+    Object.entries(countedObj).map((el) => {
+        if(el[1]%2) {
+            result = el[0]
+            return true;
+        }
+    })
```
Comment (ilya-ischenko) [inline homeworks/max-hirning/01-js/index.js:146]: 1. Map needed to create array. Why did you use it here?
2. return statement does not work in this case, you can not exit from the map function, you can put console.log inside if statement and after it to see that your loop does not exit from "if" statement.

File: homeworks/max-hirning/01-js/index.js [hunk 2]
```diff
@@ -0,0 +1,386 @@
...
+    if (distance <= moves) {
+        return "Caught!";
+    } else {
+        return "Escaped!";
+    }
+}
+
+// task 15 https://www.codewars.com/kata/duplicate-encoder
+function duplicateEncode(word) {
+    word = word.toLowerCase();
```
Comment (ilya-ischenko) [inline homeworks/max-hirning/01-js/index.js:247]: Do not mutate input argument

File: homeworks/max-hirning/01-js/index.js [hunk 3]
```diff
@@ -0,0 +1,386 @@
...
+    string.split("").map((el) => {
+        if(el !== " ") {
+            if(n === word.length) {
+                res.push(word);
+                word = el
+            } else {
+                word += el
+            }
+        }
+    });
```
Comment (ilya-ischenko) [inline homeworks/max-hirning/01-js/index.js:366]: The same situation with map, map used to create new array

**Review by ilya-ischenko — COMMENTED** [review]
1. Rename homework folder according to lecture name - 06-js-basics-1
2. In root folder of our laba we have "package.json" file with needed dependecies, make "yarn" command in root directory in order to instal packages and format your code, please.

File: homeworks/max-hirning/01-js/index.js [hunk 4]
```diff
@@ -0,0 +1,386 @@
...
+            return value1 - value2;
+        case '*':
+            return value1 * value2;
+        case '/':
+            if (value2 === 0) {
+                return "Error: Division by zero";
+            }
+            return value1 / value2;
+        default:
+            return "Error: Invalid operation";
```
[the remark this answers is not in the document]
Comment (Max-Hirning) [reply homeworks/max-hirning/01-js/index.js:21]: Fixed, forgot about it

File: homeworks/max-hirning/01-js/index.js [hunk 5]
```diff
@@ -0,0 +1,386 @@
...
+// task 4 http://www.codewars.com/kata/transportation-on-vacation
+function rentalCarCost(d) {
+    let result = d * 40;
+    if (d >= 7) {
+        result -= 50;
+    } else if (d >= 3) {
+        result -= 20;
+    }
+    return result;
+}
```
[the remark this answers is not in the document]
Comment (Max-Hirning) [reply homeworks/max-hirning/01-js/index.js:59]: Fixed. Will remember it

File: homeworks/max-hirning/01-js/index.js (hunk 1 above)
Comment (Max-Hirning) [reply homeworks/max-hirning/01-js/index.js:146]: Fixed. The reason why I used map: it doesn't mutate arr. But didn't take into account that return won't work as break in loop

File: homeworks/max-hirning/01-js/index.js (hunk 2 above)
Comment (Max-Hirning) [reply homeworks/max-hirning/01-js/index.js:247]: Fixed. Got it

File: homeworks/max-hirning/01-js/index.js (hunk 3 above)
Comment (Max-Hirning) [reply homeworks/max-hirning/01-js/index.js:366]: Fixed.
This is already a habit: I was said that it is better to use map: it runs through a copy of the array. So I used it. 
But I understood what the problem was

**Review by ilya-ischenko — COMMENTED** [review]
1. Delete 01-js, please, you should have only one folder for this homework
2. Do not make changes in package.json file, please

**Comment by Max-Hirning** [comment]
> 1. Delete 01-js, please, you should have only one folder for this homework
> 2. Do not make changes in package.json file, please

Dleted 01-js folder
Canceled changes in package.json

File: homeworks/max-hirning/01-js/index.js (hunk 1 above)
Comment (ilya-ischenko) [inline homeworks/max-hirning/01-js/index.js]: You still use map here, it generates useless array
Do not you want to use forEach for it?

File: homeworks/max-hirning/01-js/index.js (hunk 3 above)
Comment (ilya-ischenko) [reply homeworks/max-hirning/01-js/index.js]: ForEach method?

File: homeworks/max-hirning/01-js/index.js (hunk 1 above)
Comment (Max-Hirning) [reply homeworks/max-hirning/01-js/index.js]: image
 Same thing with next comment

**Pull request merged**